testProject/ViewController.m

```diff
@@ -0,0 +1,216 @@
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    self.titleImage.image = [UIImage imageWithData:[NSData dataWithContentsOfURL:[NSURL URLWithString:@"https://pbs.twimg.com/media/CAnneWrW8AAeGJE.jpg"]]];
+    self.txtboxSearch.delegate = self;
+    [self.tweetsTableView registerNib:[UINib nibWithNibName:NSStringFromClass([TweetsTableViewCell class])
+                                                     bundle:[NSBundle mainBundle]]
+               forCellReuseIdentifier:NSStringFromClass([TweetsTableViewCell class])];
+    
+    self.users = [NSMutableArray array];
+    self.tweets = [NSMutableArray array];
+    self.profilePictureUrls = [NSMutableArray array];
+    [self authenticateWithCompletion:nil];
+}
+
+#pragma mark - UITextFieldDelegate
+
+- (BOOL)textFieldShouldReturn:(UITextField *)textField {
+    [textField resignFirstResponder];
+    if (textField == self.txtboxSearch) {
+        [self.searchButton sendActionsForControlEvents:UIControlEventTouchUpInside];
+        return NO;
+    }
+    return YES;
+}
+
+
+#pragma mark - UITableViewDelegate
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+    return [self.users count];
+}
+
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
+    
+    TweetsTableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:NSStringFromClass([TweetsTableViewCell class])];
+    
+//    if(cell == nil)
+//    {
+//        cell = [self.tweetsTableView dequeueReusableCellWithIdentifier:NSStringFromClass([TweetsTableViewCell class])];
+//    }
+    
+    NSURLRequest *profileImageRequest = [NSURLRequest requestWithURL:[NSURL URLWithString:self.profilePictureUrls[indexPath.row]] cachePolicy:NSURLRequestReturnCacheDataElseLoad timeoutInterval:10];
+    
+//    AFURLSessionManager *afSessionManager = [[AFURLSessionManager alloc] init];
+    AFImageDownloader *afImageDownloader = [[AFImageDownloader alloc] init];
+    [afImageDownloader downloadImageForURLRequest:profileImageRequest
+                                          success:^(NSURLRequest *request, NSHTTPURLResponse *response, UIImage *responseObject) {
+                                              cell.userPicture.image = responseObject;
+    }
+                                          failure:^(NSURLRequest *request, NSHTTPURLResponse *response, NSError *error) {
+        // TODO: add error log
+    }];
+    cell.username.text = self.users[[indexPath row]];
+    cell.tweet.text = self.tweets[[indexPath row]];
+    
+    return cell;
+}
+
+- (void)scrollViewDidScroll: (UIScrollView *)scroll {
+    // UITableView only moves in one direction, y axis
+    CGFloat currentOffset = scroll.contentOffset.y;
+    CGFloat maximumOffset = scroll.contentSize.height - scroll.frame.size.height;
+    
+    // Change 10.0 to adjust the distance from bottom
+    if (maximumOffset - currentOffset <= 10.0) {
+        if(_pageAvailable)
```

Comment:
can get rid of this nested If with (maxiumOffset - currentOffset <= 10.0 && _pageAvailable)